monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/EvalOnStartOperator.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.internal.operators
+
+import monix.eval.Task
+import monix.execution.Ack
+import monix.execution.Ack.Stop
+import monix.reactive.Observable.Operator
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+private[reactive] final
+class EvalOnStartOperator[A](cb: A => Task[Unit])
+  extends Operator[A,A] {
+
+  def apply(out: Subscriber[A]): Subscriber[A] =
+    new Subscriber[A] {
+      implicit val scheduler = out.scheduler
+
+      private[this] var isDone = false
+      private[this] var isStart = true
+
+      def onNext(elem: A): Future[Ack] = {
+        if (isStart) {
+          // Protects calls to user code from within the operator and
+          // stream the error downstream if it happens, but if the
+          // error happens because of calls to `onNext` or other
+          // protocol calls, then the behavior should be undefined.
+          val t = try {
+            cb(elem)
+          } catch {
+            case NonFatal(ex) => Task.raiseError(ex)
+          }
+
+          isStart = false
+
+          t.attempt.flatMap {
+            case Left(ex) => Task.eval { onError(ex); Stop }
+            case Right(()) => Task.deferFuture(out.onNext(elem))
```

Comment:
We run it right away and need `Task` only for types so IMO we could use `Task.fromFuture`

Reply:
yep, makes sense, I just thought that you can't be "too lazy" in this case :D